tests/js/reactions.js

```diff
@@ -181,6 +246,72 @@ suite('Custom Element Reactions', function() {
 
   });
 
+  suite('customized built-in elements', function() {
+    test('inherits native behavior such as a button\'s disabled property', function(done) {
+      var numClickCalls = 0;
+      var isDisabled;
+
+      class CustomizedBuiltin7 extends HTMLButtonElement {
+        constructor() {
+          super();
+          this.disabled = true;
+          isDisabled = true;
+        }
+      }
+
+      customElements.define('customized-builtin-7', CustomizedBuiltin7, {extends: 'button'});
+      var button = document.createElement('button', {is: 'customized-builtin-7'});
+      var shouldBeDisabled = true;
+
+      button.addEventListener('click', function() {
+        if (isDisabled) {
+          assert.fail();
+          done();
+        } else {
+          done();
+        }
+      });
+      button.click();
+
+      setTimeout(function() {
+        isDisabled = false;
+        button.disabled = false;
+        button.click();
+      });
+    });
+
+    test('does not upgrade elements that don\'t match what extends was at definition', function() {
```

Comment:
This appears to be correct. Spec breadcrumbs: https://html.spec.whatwg.org/#concept-try-upgrade will give a null definition in step 1 since https://html.spec.whatwg.org/#look-up-a-custom-element-definition will return null if name + local name do not match (in its step 5).